### PR TITLE
Enable cross compilation with MinGW

### LINKED ISSUE #147: Problems of cross compiling with Linux and targeting Windows
OpenWSMan provides some libc functions when detecting that specific headers not found or WIN32 defined. But they're conflicting when building by MinGW. 
MinGW provides almost full glibc functions as standard library, this makes `TIME_WITH_SYS_TIME` defined and the code including such as `sys/time.h`, while `WIN32` alse defined, self-defined type like `timezone` is alse compiled. Duplicated type error occurs.

### PR DESCRIPTION
**Pull request by bgermann**
Let the client cross compile under Linux with MinGW for Windows.
This closes #147.

### CODE REVIEW

File: src/lib/wsman-curl-client-transport.c [hunk 1]
```diff
@@ -283,7 +283,10 @@ init_curl_transport(WsManClient *cl)
 	CURLcode r = CURLE_OK;
         char *sslhack;
         long sslversion;
-        dictionary *ini = iniparser_new(wsmc_get_conffile(cl));
+        dictionary *ini = NULL;
+#ifndef _WIN32
+        ini = iniparser_new(wsmc_get_conffile(cl));
```
Comment (kkaempf) [inline src/lib/wsman-curl-client-transport.c:288]: why is the initialization ifdef'ed out for Windows ?

**Comment by kkaempf** [comment]
This is mixing at least three different changes and thus should be splitted into individual commits:

- removal of `rcsid`s
- declaration of `wsmc_set_selectors_from_uri`
- MinGW enablement

the `strftime` change and the `bzero` to `memset` switch should also be separated commits.

File: src/lib/wsman-curl-client-transport.c (hunk 1 above)
Comment (bgermann) [reply src/lib/wsman-curl-client-transport.c:288]: The commit title should actually explain it:  "Hide wsmc_get_conffile behind _WIN32 (like prototype)". With commit 9ec439aa18efa48cce3eaa999f9b9742795ee60b this line was introduced without changing the wsmc_get_conffile prototype to be available with _WIN32. From there, I do not think the initialization compiled on Windows. This should be fixed but I do not know your configuration architecture and do not know why the .h has the ifdefs as they are.

**Comment by bgermann** [comment]
Actually, they are already in separate commits. There are six commits in this pull request. I guess you lokked at them in bulk.

File: src/lib/wsman-curl-client-transport.c (hunk 1 above)
Comment (kkaempf) [reply src/lib/wsman-curl-client-transport.c:288]: But where do you call `iniparser_new` now ?

Comment (bgermann) [reply src/lib/wsman-curl-client-transport.c:288]: Nowhere. This makes the curl client transport compile on Windows (or cross compile for it) but does not fix the initialization which did not even compile before. So even though this is still non-functional it at least compiles.

**Comment by kkaempf** [comment]
Thanks !

**Pull request merged**